### ISSUE TITLE
link to official repo at the top of readme

### ISSUE THREAD

**Issue by ysymyth** [kyegomez/tree-of-thoughts#56]
hi @kyegomez , it is absolutely not cool to close this issue twice without any action. #54 #55 

you have to acknowledge that you are not the official implementation, and link to the official implementation at the top of README.md. Otherwise, when you implemented something wrong, people will be misled and think ToT is trash, and jeopardize the reputation of our research.

If you close again, I would have to report to GitHub to close down this content via https://docs.github.com/en/site-policy/content-removal-policies. thanks in advance.

**Comment by kyegomez** [comment]
Nope, it's absolutely not cool to let your ego control your emotions.

You seem to be more interested in gaining credit for code that is nowhere similar than yours than actually advancing the algorithm, this is VERY not cool and this is why I made this repository in the first place. Way too many people just politicking and wasting time instead of investing time and energy into activities that matter.

And, I don't have to do anything, I've acknowledged you and your team twice. Now stop acting like a child and get back to work on things that matter. And, you cannot do that because our code is NOTHING like yours, you'd just waste your time and waste the peoples time.

**Comment by ysymyth** [comment]
you don't seem to get the message. I was not trying to close or discredit your repo, but just merely asking you to link our repo into your readme, and I do not understand why it is so hard, given it's not harming your repo in any way.

and it's not about the credit, it's about you might implement something wrong and mislead users into thinking "Tree of Thoughts" is trash. Without linking to our repo, any user that googles "Tree of Thoughts" and goes to your repo and gets disappointed will blame the paper as trash. 

you leave me no option but to report to Github and seek legal actions.

**Comment by kyegomez** [comment]
Why do I need to link your repo? There is no need if I do not choose to.

And this is hypothetical situation. And what if your repo is "trash" and gets disappointed and thinks tree of thoughts is trash? I helped grow this idea you should be more thankful, but you let your emotions cloud your judgement.

And, if you even think about closing our repo you're an authoritarian monkey whose only devoted to himself and not the advancement of Humanity, as leaving it up is open source. An freedom of expression.

And you seem to think that the idea of tree of thoughts is somehow yours, it's not. You will never own anything especially information and any attempt to believe that you can you are just deceiving yourself.

And, do as you please but ask yourself: "Do I want to solve problems or create them"

**Comment by ysymyth** [comment]
😂

given you have a lot of thoughts, why not just open these issues so that people can realize your good thoughts

**Comment by off99555** [comment]
> Now stop acting like a child and get back to work

It's not cool to tell adults that they act like a child when you act like a child yourself. And you are also a child according to your profile saying 19 years old. I don't want to do ad hominem stuff here but if original author of the paper asks you to kindly link to the official repo, you should do it. You have the skills but you don't have the right attitude. Fix it.

When I was young, I used to take other people's work, improved it, and pretended it was mine as well (by phrasing words in a misleading way in the documentation to make it sound like I made the whole thing myself). You seem to be doing the same mistake like me in the past. The thing is: Young brains do mistakes. So you are not unique in this problem. It is not the end of the world if you would backtrack right now.

It's up to you now what you will do. The ball is in your court. Make it count.

**Comment by strandline** [comment]
Kyle, 

The algorithm and approach were developed by a large group of academic researchers from Princeton. Shunyu Yao and Dian Yu and Jeffrey Zhao and Izhak Shafran and Thomas L. Griffiths and Yuan Cao and Karthik Narasimhan. You created a piece of derivative software, but are not the originator of the concept. Tree of Thought. Shunyu has politely asked you to provide an appropriate reference to the original repo, and you should do that. It is the right thing to do. It is very poor behavior to not provide proper attribution as requested by authors. 

>> And you seem to think that the idea of tree of thoughts is somehow yours, it's not. You will never own anything especially information and any attempt to believe that you can you are just deceiving yourself.

It's easy to make a claim like that when you didn't do any of the work to actually create or validate the concept. Uh, yes, the ToT idea originated from the group at Princeton. They developed the original idea, tested the concept -- which requires a lot of work -- and did all of the hard work to write up a manuscript. And they provided source code for everyone to use. They are the authors, so yah, they actually are the owners of the concept. They generously released these ideas into the wild under the MIT license to allow you and others to use it for free.  

Be nice. Do the right thing.

**Comment by kyegomez** [comment]
I have referenced the authors original implementation in the readme 😃
